### PR TITLE
tinyexr: patch in build() instead of source() + link with pthread

### PR DESCRIPTION
**Pull request by prince-chrismc**
when using with_thread on Linux

Specify library name and version:  **tinyexr/1.0.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by prince-chrismc — COMMENTED** [review]
What's worse/better?
- downloading sources for each build 
*or*
- having to copy the source to the build folder?

In order to "patch in package", it requires 
```diff
-    no_copy_source = True
```
Which I am not sure the rule for this is well documented :thinking: Perhaps a follow up from #3141

File: recipes/tinyexr/all/conanfile.py [hunk 1]
```diff
@@ -59,6 +56,8 @@ def _extracted_license(self):
         return "\n".join(license_content)
 
     def package(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
```
Comment (SpaceIm) [inline recipes/tinyexr/all/conanfile.py:60]:
```suggestion

    def build():
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
            tools.patch(**patch)

    def package(self):
```

Comment (prince-chrismc) [reply recipes/tinyexr/all/conanfile.py:60]: agreed

File: recipes/tinyexr/all/conanfile.py [hunk 2]
```diff
@@ -58,6 +55,10 @@ def _extracted_license(self):
             license_content.append(content_lines[i][:-1])
         return "\n".join(license_content)
 
+    def build():
```
Comment (prince-chrismc) [inline recipes/tinyexr/all/conanfile.py:58]:
```suggestion
    def build(self):
```

**Review by SSE4 — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**